tests/functional/i/invalid/invalid_name/invalid_name_with_final_typing.py

```
@@ -0,0 +1,26 @@
+"""Regression test for https://github.com/pylint-dev/pylint/issues/10711."""
+
+# pylint: disable=missing-class-docstring, missing-function-docstring
+
+from dataclasses import dataclass
+from typing import Final
+
+module_snake_case_constant: Final[int] = 42  # [invalid-name]
+MODULE_UPPER_CASE_CONSTANT: Final[int] = 42
+
+
+def function() -> None:
+    function_snake_case_constant: Final[int] = 42  # [invalid-name]
+    FUNCTION_UPPER_CASE_CONSTANT: Final[int] = 42
+    print(function_snake_case_constant, FUNCTION_UPPER_CASE_CONSTANT)
+
+
+@dataclass
+class Class:
+    class_snake_case_constant: Final[int] = 42  # [invalid-name]
+    CLASS_UPPER_CASE_CONSTANT: Final[int] = 42
+
+    def method(self) -> None:
+        method_snake_case_constant: Final[int] = 42  # [invalid-name]
```

I suggest adding a test case for

```
self.CLASS_UPPER_CASE_ATTRIBUTE = 36 # [invalid-name]
```

Because this attribute is named with `UPPER_CASE` letters, it should not be re-assigned a new value by any means. *If* it is re-assigned a new value, we can safely raise `[invalid-name]` on the spot and asking for `snake_case` names instead.